x-pack/legacy/plugins/actions/README.md

```diff
@@ -23,6 +23,27 @@ action types.
+| _xpack.actions._**enabled**          | ?                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | boolean               |
+| _xpack.actions._**WhitelistedHosts** | Which _hostnames_ are whitelisted for the Built-In-Action? This list should contain hostnames of every external service you wish to interact with using Webhooks, Email or any other built in Action. Note that you may use the string "\*" (instead of the array) to enable Kibana to target any URL, but keep in mind the potential use of such a feature to execute [SSRF](https://www.owasp.org/index.php/Server_Side_Request_Forgery) attacks from your server. | "\*" or Array<String> |
+
+### Configuration Utilities
+
+This module provides a Utilities for interacting with the configuration.
+
+| Method                | Arguments                                          | Description                                                                                                                                                                                                                                                            | Return Type                                                                                                                                                                                                                   |
+| --------------------- | -------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| isWhitelistedHostname | _uri_: The URI you wish to validate is whitelisted | Validates whether the URI is whitelisted. This checkes the configuration and validates that the hostname of the URI is in the list of whitelisted Hosts. If the configuration says that all URI's are whitelisted (using an "\*") then it will always returns an _Ok_. | Result<String,String>: Returns a [Result type](./server/builtin_action_types/lib/result_type.ts) where an _Ok_ is returns with the whitelisted URI (if it is whitelisted) or an _Err_ with an error message if it isn't whitelisted. |
```

Comment:
Using result type for the result of this function seems like overkill; can't it just be a boolean?  I'm wondering if the current pattern we have for config/secrets/params validation would work - the function returns null-ish for "ok", and a string to be used as an error message on error.  Weird, but has worked so far.

Reply:
It used to be a boolean - I changed it to accommodate for the suggestion that this method return the error message.

I have to admit that I dislike functions that don't have a return value... it makes the behaviour far more implicit, harder to compose and makes it harder to write functional code as opposed to imperative. I accept that's subjective, but I'd rather keep it that way unless it poses a problem.
I also have to admit I fail to see what makes it "overkill"... the type expresses what the behaviour is, this seems like a positive.

Perhaps I'm just too used to functional languages, but this seems like a relatively normal pattern to me.

Reply:
In terms of "overkill", I was curious if these kinds of "tagged" "option" patterns in TS managed to elide the actual tag in the compiled output.  Didn't think they could, but ... wasn't sure.

I tried pasting the `result_type.ts` into [typescript playground](https://www.typescriptlang.org/play/index.html) and got this: [generated JS from `result_type.ts`](https://www.typescriptlang.org/play/index.html#code/KYDwDg9gTgLgBDAnmYcDyBrAPAFQHxwC8cA3gFBwICGA5gFxwDkEGjA3BXAG5UA2ArsAY4OAXw6hIsBMlQBRKFCxyCxcpRi0GjYIvaddUaAzliJ4aPCQo4AJWABnfrxi4ANHBVF02fHAA+norKeBxkkpZwAGb8AHYAxjAAlhCxcFQOmLh4ABQ8AkJwOACUDFl+6nBQwDD8UGmVGlpMLIxunJT5gu2U4mSiZOEW0jEJyanpDgpK+DmGxkWlQTMEldW19aQd1PRMhm3b81A9cH0DQ1Lwo4kpaUmZvh4qOdVOLgz2b644T3hLr854PcfNktpR1nU0gCXAA6TQ0IiEYjMVhiQYREZxG4Te7TdyeXLQmAfRyA-Eqf6klxwYF4rxrGqQuAAQnumBeVJgxTRF0i13GaTiDioUWAAHEauTCZySV8pUscGCaVE4DlcYoOV9isUlRoABZGADucFiwGN02gOQABgASEhEmFHURW7mcAbgxmbB1dYA8jHwDKIBLRLECuBgIwAW3uwE+ZJ+BJyMQ2hQAClGY9klumINGHMAsHGXFLVpwYFBELqqp60hl2VRDVQkld+CnXb04PEqDB4nrVcAdY1qxta1MNQOOL1+kA)

And at least in this case (and I suspect all similar cases), the tags do end up showing up in the generated JS, which is not pleasing to me.  The function `isWhitelistedHostname()` will have to be documented so a JS programmer could use it, and so we'd have to expose this particular implementation detail (the `tag` field), which doesn't seem right.  This particular case shape is so typical, I wonder if there's a common flavor in use somewhere else in Kibana, for other "published" APIs?  If not, seems like either the validator shape (return `void | string` (or whatever), or having the method indicate it throws an error when the argument is not whitelisted, seem more appropriate.

This is basically the same concern I had with exposing fp-ts in our APIs.  Using this kind of stuff internally, as an implementation detail, seems fine.  When it ends up showing up in exposed APIs customers may end up consuming, I get twinge-y.

Reply:
I see. I guess I didn't think of this as external...

I understand your concern, but I guess ideally we'd be able to present our APIs as Typescript. 
If our APIs have to be JS, then I fear we're never going to be able to enjoy typesafety in Kibana as we'll constant be forced to strip away our types between parts of Kibana. 

Anyway, I'll change the API.... 🤷‍♂️

Reply:
Removed the Result type, replaced with returning an Error, which what the typescript docs recommend in place of throwing an error.